tests/e2e/datadog/logs/mod.rs

```diff
@@ -0,0 +1,69 @@
+use serde_json::Value;
+
+use vector::test_util::trace_init;
+
+use super::*;
+
+const LOGS_ENDPOINT: &str = "/api/v2/logs";
+
+fn expected_log_events() -> u32 {
+    std::env::var("EXPECTED_LOG_EVENTS")
+        .unwrap_or_else(|_| "1000".to_string())
+        .parse::<u32>()
+        .expect("EXPECTED_LOG_EVENTS should be an unsigned int")
+}
+
+// Asserts that each log event has the hostname and timestamp fields, and
+// Removes them from the log so that comparison can more easily be made.
+// @return the number of log entries in the payload.
+fn assert_timestamp_hostname(payloads: &mut Vec<Value>) -> u32 {
```

Comment:
nit: Why `u32` and not `usize`? Also, since you're not modifying the `Vec` itself (i.e. adding or removing elements), just its contents, I think the payload could be `&mut [Value]`.